runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -53,16 +52,8 @@ protected ILogData removeFromQueue(NavigableSet<Long> queue) {
             }
 
             thisRead = queue.pollFirst();
-            try {
-                ld = read(thisRead);
-            } catch (TrimmedException te) {
-                if (!getReadOptions().isIgnoreTrim()) {
-                    throw te;
-                }
 
-                readNext = true;
-                continue;
-            }
+            ld = read(thisRead);
```

Comment:
shouldn't you check if the read entry is compacted? in case it is compacted set readNext to true and go to next log entry, as current one is not valid.

Reply:
Yes. This is a issue! I plan to eliminate the backpointerStreamView in a dedicated patch.

Reply:
You should also check compacted for bitmap implementation.

Reply:
I've checked that read() is compatible with bitmap implementation.

Reply:
I mean check if the address is compacted for bitmap stream layer implementation.

Reply:
If this is a issue, I will address it in the branch called compactionMarkInRuntime.